cdap-docs/reference-manual/source/release-notes.rst

```diff
@@ -17,12 +17,663 @@
 Cask Data Application Platform Release Notes
 ============================================
 
+.. New Features
+.. Improvements
+.. Bug Fixes
+.. Known Issues
+.. API Changes
+.. Deprecated and Removed Features
+
 .. contents::
    :local:
    :class: faq
    :backlinks: none
    :depth: 2
 
+`Release 4.0.0 <http://docs.cask.co/cdap/4.0.0/index.html>`__
+=============================================================
+
+New Features
+------------
+
+#. Cask Market
+
+   - :cask-issue:`CDAP-7203` - Adds the new Cask Market view in the CDAP UI for users to
+     interact and create CDAP entities in a much easier fashion. The market hosts use cases,
+     pipelines, datapacks, drivers, Hydrator plugins, and CDAP applications as examples.
+
+#. Cask Wrangler
+
+   - :cask-issue:`WRANGLER-2` - Added Cask Wrangler: a new CDAP extension for data
+     preparation.
+
+#. CDAP Transactional Messaging System
+
+   - :cask-issue:`CDAP-7211` - Adds a transactional messaging system that can be used to
+     communicate between components.
+
+#. Dynamic Log Level
+
+   - :cask-issue:`CDAP-5479` - Allow updating or resetting of log levels for program types
+     worker, flow, and service dynamically using REST endpoints.
+ 
+   - :cask-issue:`CDAP-7214` - Allow setting the log levels for all program types through
+     runtime arguments or preferences.
+
+#. Operational Statistics
+
+   - :cask-issue:`CDAP-7670` - Added support for reporting of operational statistics from
```

Comment:
Added a pluggable extension to get operational statistics in CDAP. Provided extensions to get operational stats from YARN, HDFS, HBase and CDAP.